src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -39,7 +34,19 @@ public class BlockStoreSignaled : SignalObserver<Block>
         private readonly IBlockStoreCache blockStoreCache;
 
         /// <summary>Queue of chained blocks that will be announced to the peers.</summary>
-        private readonly ConcurrentQueue<ChainedBlock> blocksToAnnounce;
+        private AsyncQueue<ChainedBlock> blocksToAnnounce;
+
+        /// <summary>Local batch of blocks that will be announced to the peers.</summary>
+        private List<ChainedBlock> localBatch;
```

Comment:
this and the queue readonly?

Reply:
They are created in Initialize().
I can move creation to the constructor and make them readonly but that doesn't sound very logical to me- they belong to Initialize() method.

Reply:
Fixed.